kedro-datasets/kedro_datasets/matlab/matlab_dataset.py

```diff
@@ -0,0 +1,157 @@
+"""``MatlabDataset`` loads/saves data from/to a Matlab file using an underlying
+filesystem ?(e.g.: local, S3, GCS)?. The underlying functionality is supported by
+the specified backend library passed in (defaults to the ``matlab`` library), so it
+supports all allowed options for loading and saving matlab files.
+"""
+import numpy as np
+import warnings
+from copy import deepcopy
+from scipy import io
+from pathlib import PurePosixPath
+from typing import Any, Dict
+
+import fsspec
+from kedro.io.core import Version, get_filepath_str, get_protocol_and_path
+
+from kedro_datasets import KedroDeprecationWarning
+from kedro_datasets._io import AbstractVersionedDataset, DatasetError
+
+
+class MatlabDataset(AbstractVersionedDataset[np.ndarray, np.ndarray]):
+    """`MatlabDataSet` loads and saves data from/to a MATLAB file using scipy.io.
+    
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog_yaml_examples.html>`_:
+
+    .. code-block:: yaml
+        cars:
+            type: mat.MatlabDataset
+            filepath: gcs://your_bucket/cars.mat
+            fs_args:
+                project: my-project
+            credentials: my_gcp_credentials
+    
+    Example usage for the
+    `Python API <https://kedro.readthedocs.io/en/stable/data/\
+    advanced_data_catalog_usage.html>`_:
+
+    .. code-block:: pycon
+        >>> from kedro_datasets.matlab import MatlabDataset
+        >>> data = {"col1": [1, 2], "col2": [4, 5], "col3": [5, 6]}
+        >>> dataset = MatlabDataset(filepath='/data/my_data.mat')
+        >>> dataset.save(data)
```

Comment:
Looks like this fails:

```
[Errno 13] Permission denied: '/data/my_data.mat'
```

probably because `/data` doesn't exist. Maybe let's not use any directories?

```suggestion
        >>> dataset = MatlabDataset(filepath="my_data.mat")
        >>> dataset.save(data)
```

Reply:
I managed to get "make link plugin=kedro-datasets" to work and I get this message:

> (kedro) PS G:\Working\kedro-plugins\kedro-plugins\kedro-plugins> make lint plugin=kedro-datasets
pre-commit run -a --hook-stage manual ruff-kedro-datasets && pre-commit run trailing-whitespace --all-files && pre-commit run end-of-file-fixer --all-files && pre-commit run check-yaml --all-files && pre-commit run check-added-large-files --all-files && pre-commit run check-case-conflict --all-files && pre-commit run check-merge-conflict --all-files && pre-commit run debug-statements --all-files && pre-commit run black-kedro-datasets --all-files --hook-stage manual && pre-commit run secret_scan --all-files --hook-stage manual && pre-commit run bandit --all-files --hook-stage manual
Ruff on kedro_datasets/*.................................................Failed
- hook id: ruff-kedro-datasets
- exit code: 1
- files were modified by this hook

Found 10 errors (10 fixed, 0 remaining).

make: *** [Makefile:17: lint] Error 1